src/NewTwitchApi/Resources/AbstractResource.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NewTwitchApi\Resources;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Psr7\Request;
+use NewTwitchApi\RequestResponse;
+
+abstract class AbstractResource
+{
+    protected $guzzleClient;
+
+    public function __construct(Client $guzzleClient)
+    {
+        $this->guzzleClient = $guzzleClient;
+    }
+
+    protected function callApi(string $uriEndpoint, array $queryParamsMap = [], string $bearer = null): RequestResponse
+    {
+        $request = new Request(
+            'GET',
+            sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+            $bearer ? ['Authorization' => sprintf('Bearer %s', $bearer)] : []
+        );
+        try {
+            $response = $this->guzzleClient->send($request);
+        } catch (GuzzleException $e) {
```

Comment:
Why do you catch this? Shouldn't we at least propagate exception?

Reply:
Hm. Possibly. I'll look into that.